analyzer/windows/modules/auxiliary/autoruns.py

```diff
@@ -0,0 +1,145 @@
+            autoruns = os.path.join(bin_path, "autorunsc64.exe")
+        else:
+            autoruns = os.path.join(bin_path, "autorunsc.exe")
+
+        if not os.path.exists(autoruns):
+            raise CuckooPackageError(
+                "In order to use the Autorun functionality, it "
+                "is required to have Autorunsc setup with Cape."
+            )
+        autoruns = autoruns.replace("\\","\\\\")
+        run_args = self.options.get("run_args")
+        if not run_args:
+            run_args = f"-a * -c -o {self.output_dir}\\\\{self.output_file_start}"
+
+        run_cmd = f"{autoruns} {run_args}"
+        run_cmd = shlex.split(run_cmd)
+        log.debug(run_cmd)
+        #with open(f"{self.output_dir}\\{self.output_file_start}", "w") as f:
+        #    subprocess.Popen(run_cmd, startupinfo=self.startupinfo, stdout=f, stderr=f, text=True)
+        subprocess.Popen(run_cmd, startupinfo=self.startupinfo)
+
+    def stop(self):
+
+        # First figure out what architecture the system in running (64 or 86)
+        bin_path = os.path.join(os.getcwd(), "bin")
+
+        if "AMD64" in platform.uname():
+            autoruns = os.path.join(bin_path, "autorunsc64.exe")
+        else:
+            autoruns = os.path.join(bin_path, "autorunsc.exe")
+
+        if not os.path.exists(autoruns):
+            raise CuckooPackageError(
+                "In order to use the Autorun functionality, it "
+                "is required to have Autorunsc setup with Cape."
+            )
+        autoruns = autoruns.replace("\\","\\\\")
+        run_args = self.options.get("run_args")
+        if not run_args:
+            run_args = f"-a * -c -o {self.output_dir}\\\\{self.output_file_end}"
+
+        run_cmd = f"{autoruns} {run_args}"
+        run_cmd = shlex.split(run_cmd)
+        log.debug(run_cmd)
+        #with open(f"{self.output_dir}\\{self.output_file_end}", "w") as f:
+        #    process = subprocess.Popen(run_cmd, startupinfo=self.startupinfo, stdout=f, text=True)
+        process = subprocess.Popen(run_cmd, startupinfo=self.startupinfo)
+        process.wait()
+        
+        start_elements = []
+        end_elements = []
+        diff_elements = []
+        path_start = os.path.join(self.output_dir, self.output_file_start)
+        path_end = os.path.join(self.output_dir, self.output_file_end)
+        path_diff = os.path.join(self.output_dir, self.output_file_diff)
+
+        with open(path_start, mode='r') as f:
+            reader = csv.DictReader(f, delimiter=",")
+            for row in reader:
+                start_elements.append(row)
+        with open(path_end, mode='r') as f:
+            reader = csv.DictReader(f, delimiter=",")
+            for row in reader:
+                keys = row.keys()
+                end_elements.append(row)
+
+        for element in start_elements:
+            if element not in end_elements:
+                diff_elements.append(f"--,{element}")
```

Comment:
hmm this would be really interesting actually. if a malware removes a default autorun....